### PR TITLE
[Feature] Support weight-shape-unaligned block-scale fp8 models

### PR DESCRIPTION
**Pull request by Wanli-Jiang**
## Purpose

- Support Nemotron-3-Nano block-scale fp8 models for now.

## Test Plan

* Step 1: Generate block-scale fp8 model for Nemotron-3-Nano
```
(using [modelopt](https://github.com/NVIDIA/Model-Optimizer) to quantize bf16 model.

# CMD:
python examples/llm_ptq/hf_ptq.py \
--pyt_ckpt_path nvidia/NVIDIA-Nemotron-3-Nano-30B-A3B-BF16 \
--qformat fp8_pb_wo \
--kv_cache_qformat none \
--export_path Nemotron-3-Nano-FP8-pb-wo \
--trust_remote_code
```

* Step 2: Test it with GPQA-diamond / LCB_v6 and compare with bf16 model ckpt.

## Test Result

* GPQA-diamond:  71.82@bf16, 70.2@fp8-blockscale.
* LCB_v6: 64.36@bf16, 66.74@fp8-blockscale.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/modelopt.py [hunk 1]
```diff
@@ -822,47 +864,127 @@ def create_weights(
...
+        )
+        layer.w2_weight_scale = Parameter(
+            layer.w2_weight_scale.squeeze(), requires_grad=False
+        )
+
+        # Apply padding if needed
+        required_padding = layer.w13_weight_scale.size(-2) * block_shape[
+            0
+        ] - w13_weight.size(-2)
+        if required_padding != 0:
```
[the remark this answers is not in the document]
Comment (Wanli-Jiang) [reply vllm/model_executor/layers/quantization/modelopt.py]: The layer.w13_weight_scale is squeezed as `[num_experts, out_blocks, in_blocks]`. So that  it is correct.

File: vllm/model_executor/layers/quantization/modelopt.py [hunk 2]
```diff
@@ -822,47 +864,127 @@ def create_weights(
...
+                torch.nn.functional.pad(w13_weight, (0, 0, 0, required_padding)),
+                requires_grad=False,
             )
-            w2_input_scale = PerTensorScaleParameter(
-                data=torch.full((num_experts,), 1.0, dtype=torch.float32),
-                weight_loader=weight_loader,
+            layer.w2_weight = Parameter(
+                torch.nn.functional.pad(w2_weight, (0, required_padding, 0, 0)),
+                requires_grad=False,
             )
```
[the remark this answers is not in the document]
Comment (Wanli-Jiang) [reply vllm/model_executor/layers/quantization/modelopt.py:978]: accept the suggestion.

**Comment by Wanli-Jiang** [comment]
> @Wanli-Jiang we don't need this padding and logic in the linear layer to support block-unaligned models in other backends such as fp8.py and compressed-tensors due to the kernels that support it. Is this truly needed for modelopt?

@mgoin thanks for your reply!
 
For modelopt, at least I need such code changes to make it loading models correctly and run well with cutlass FP8 linear.

can you point me out the logic to suport block-unaligned models in fp8.py? I did not see any special handling for that actually. Thanks!
